atlas/AtlasMl/atlasml/adk_tools.py

```
@@ -0,0 +1,187 @@
+        artemis_client = _get_artemis_client()
+        exercises = artemis_client.get_exercises(course_id)
+        if exercises:
+            return artemis_client.format_exercises_for_display(exercises)
+        else:
+            return f"No exercises found for course {course_id}."
+    except Exception as e:
+        logger.error(f"Failed to get exercises: {str(e)}")
+        return f"Failed to get exercises: {str(e)}"
+
+'''@adk_tool
+def apply_competency_mapping(competency_id: str, exercise_id: int, course_id: str) -> str:
```

Ruff (0.12.2)

Line too long (90 > 88)

(E501)